### DISCUSSION TITLE
Adding User-Agent header to IPnPContextFactory.CreateAsync()

### DISCUSSION THREAD

**Discussion by Amwins-Bass-Guy** [pnp/pnpcore#1551] [Q&A]
We are trying to verify that all of our PnP requests have our specified `User-Agent` header to make sure that Microsoft can track down any throttling (or other issues) by client application.  Where we can, all PnP calls have `.WithHeaders()` specified, but, when the PnPContext is created (using `CreateAsync()`), those two requests have the default user agent value that was set in `AddPnPCore()`.

Is there a way to pass a request header using the `PnPContextOptions` (or other mechanism), to `CreateAsync()`?

![image](https://github.com/user-attachments/assets/75fa8cc6-9119-47e4-81a4-f281ee5b079b)

Default UserAgent:
![image](https://github.com/user-attachments/assets/a01906b6-9131-4c18-b15c-6f1cdd7cec8f)

**Comment by jansenbe** [comment]
@Amwins-Bass-Guy : the ask is not fully clear to me, you're able to set a default user-agent for your application using the configuration options, but for some calls you want to set another one. Today you use the .WithHeaders option for that, but you'd rather override the user-agent when creating the PnPContext?

**Reply by Amwins-Bass-Guy** [reply]
We are trying to make sure that all calls to Sharepoint have a User-Agent that matches the particular client that is making the call.  The only calls that are using the default User-Agent from the config are the one or two calls that each CreateAsync() makes for the PnPContext.  It would be nice to be able to have something similar to .WithHeaders() for CreateAsync() - or one of the optional parameters for CreateAsync() could add the User-Agent header specifically.
